petab/v1/core.py

```diff
@@ -299,6 +300,10 @@ def flatten_timepoint_specific_output_overrides(
     petab_problem.observable_df.index.name = OBSERVABLE_ID
     petab_problem.measurement_df = pd.concat(new_measurement_dfs)
 
+    # remove visualization df if it is invalid
+    if not validate_visualization_df(petab_problem):
```

Comment:
The docs for `validate_visualization_df` state
> Returns: `True` if errors occurred, `False` otherwise

```suggestion
    if validate_visualization_df(petab_problem):
```